Engine/AlgorithmManager.cs

```diff
@@ -1256,7 +1271,13 @@ private void SampleBenchmark(IAlgorithm algorithm, IResultHandler results, DateT
         /// </summary>
         private static bool EndTimeIsInNativeResolution(SubscriptionDataConfig config, DateTime dataPointEndTime)
         {
-            if (config.Increment == TimeSpan.Zero)
+            if (config.Resolution == Resolution.Tick
+                ||
+                // time zones don't change seconds or milliseconds so we can
+                // shortcut timezone conversions
+                (config.Resolution == Resolution.Second
+                || config.Resolution == Resolution.Minute)
```

Comment:
I'm pretty sure this is safe -- for a moment I was contemplating the affect of leap seconds, but IIRC leap seconds are applied to UTC and all other time zones use UTC as a reference, so we should be good.